homeassistant/components/iotawatt/sensor.py

```diff
@@ -0,0 +1,133 @@
+"""Support for IoTaWatt Energy monitor."""
+import logging
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+)
+from homeassistant.const import (
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_WATT_HOUR,
+    POWER_WATT,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import entity_registry
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import IotaWattEntity
+from .const import COORDINATOR, DOMAIN, SIGNAL_ADD_DEVICE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add sensors for passed config_entry in HA."""
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
+    entities = []
+
+    # pylint: disable=unused-variable
+    for idx, ent in enumerate(coordinator.data["sensors"]):
+        entity = IotaWattSensor(
+            coordinator=coordinator,
+            entity=ent,
+            mac_address=coordinator.data["sensors"][ent].hub_mac_address,
+            name=coordinator.data["sensors"][ent].getName(),
+        )
+        entities.append(entity)
+
+    async_add_entities(entities)
+
+    async def async_new_entities(sensor_info):
+        """Remove an entity."""
+        ent = sensor_info["entity"]
+        hub_mac_address = sensor_info["mac_address"]
+        name = sensor_info["name"]
+
+        entity = IotaWattSensor(
+            coordinator=coordinator,
+            entity=ent,
+            mac_address=hub_mac_address,
+            name=name,
+        )
+        entities = [entity]
+        async_add_entities(entities)
+
+    async_dispatcher_connect(hass, SIGNAL_ADD_DEVICE, async_new_entities)
+
+
+class IotaWattSensor(IotaWattEntity):
+    """Defines a IoTaWatt Energy Sensor."""
+
+    def __init__(self, coordinator, entity, mac_address, name):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator, entity=entity, mac_address=mac_address, name=name
+        )
+
+        sensor = self.coordinator.data["sensors"][entity]
+        self._ent = entity
```

Comment:
This param needs a better name since this is the entity class. It looks like it is better named `key`?

Reply:
@balloob I renamed the parameter ```entity``` to ```key```. Is this what you had in mind?